cpu/stm32f4/startup.c

```diff
@@ -41,6 +41,11 @@ extern void kernel_init(void);
 extern void __libc_init_array(void);
 
 /**
+ * Required by g++ cross compiler
+ */
+void* __dso_handle;
```

Comment:
@OlegHahm the `void *__dso_handle;` is needed for destructors to provide the pointer to a handle for the DSO (Dynamic Shared Object) [1]. Unfortunately  I have no further information at hand now.

> isn't required for other platforms as well?

from my current state of knowledge yup, all platforms that are supposed to use the RIOT-STL wrapper will need this. (At least the ones compiled with `g++`)

[1] http://wiki.osdev.org/C%2B%2B#GCC